azkaban-common/src/test/java/azkaban/executor/InteractiveTestJob.java

```diff
@@ -137,6 +138,12 @@ public void succeedJob(final Props generatedProperties) {
     }
   }
 
+  public void ignoreCancel() {
+    synchronized (this) {
+      this.ignoreCancel = true;
```

Comment:
Want me to change `ignoreCancel` to volatile? Somehow I would be inclined to prefer the sync block because that it can be maybe easier to understand this class as a whole? When you don't need to pay special attention to that special volatile field and how it differs from those other flags.. Well, that refactoring to change all of them is still possible. Then it would be just the waits and the notifys that need to be inside a sync block.